File: .ipynb_checkpoints/linear_model-checkpoint.r
library(ggplot2)

dataset <- read.csv("regression_data.csv")

#data
df <- data.frame(
  x = dataset$YearsExperience,
  y = dataset$Salary)

#fit model
model <- lm(y ~ x, data = df)
slope <- coef(model)[2]
intercept <- coef(model)[1]
r <- cor(df$x, df$y)
pred <- predict(model)
mse <- mean((df$y - pred)^2)

#plot
ggplot(df, aes(x = x, y = y)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE, color = "red") +
  annotate("text", x = 1.5, y = max(df$y) - 0.5,
           label = paste("y =", round(slope, 2), "x +", round(intercept, 2),
                         "\nr =", round(r, 2), "\nMSE =", round(mse, 2)),
           size = 4) +
  labs(title = "Linear Fit of Salary vs Experience",
       x = "Years of Experience", y = "Salary") +
  theme_minimal()

ggsave("regression_plot_r.png")

File: .ipynb_checkpoints/linear_model-checkpoint.py
import sys
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import linregress
from sklearn.metrics import mean_squared_error

dataset = pd.read_csv("regression_data.csv")

# Data
x = dataset["YearsExperience"]
y = dataset["Salary"]

# Linear regression
slope, intercept, r_value, p_value, std_err = linregress(x, y)
y_pred = slope * x + intercept
mse = mean_squared_error(y, y_pred)

# Plot
plt.scatter(dataset["YearsExperience"], dataset["Salary"], color="blue")
plt.plot(x, y_pred, 'r-', label='Fitted Line')
plt.text(1.25, 55000,
         f"y = {slope:.2f}x + {intercept:.2f}\n"
         f"r = {r_value:.2f}\nMSE = {mse:.2f}",
         fontsize=12)
plt.xlabel("Years Experience")
plt.ylabel("Salary")
plt.title("Linear Regression Model of Years Experience vs Salary")
plt.legend()
plt.savefig("regression_plot_python.png")
plt.show()



